File: divide-array-into-equal-pairs.ts
{
    function divideArray(nums: number[]): boolean {
        const hash = new Map();

        for(const num of nums) {
            const new_value = (hash.get(num) || 0) + 1;

            hash.set(num, new_value)
        }

        for(const num of nums) {
            if(hash.get(num) % 2 !== 0) 
                return false
        }

        return true
    };

    console.log(divideArray([1,2,3,4]))
}

File: divide-array-into-equal-pairs.rs
use std::collections::HashSet;

pub fn divide_array(nums: Vec<i32>) -> bool {
    let mut set = HashSet::new();

    for num in nums {
        if !set.contains(&num) {
            set.insert(num);
        } else {
            set.remove(&num);
        }
    }

    return set.len() == 0;
}
